sklearn/multioutput.py

```diff
@@ -427,7 +427,15 @@ class _BaseChain(BaseEstimator, metaclass=ABCMeta):
     def __init__(self, base_estimator, *, order=None, cv=None,
                  random_state=None):
         self.base_estimator = base_estimator
-        self.order = order
+
+        if order == 'random' or order is None:
+            self.order = order
+        else:
+            self.order = np.array(order)
+            if len(self.order.shape) != 1:
+                raise ValueError(f'order is {order}, must be 1-dimensional, '
+                                 '"random", or None.')
+
```

Comment:
There should not be logic nor input validation in the `__init__`. The corresponding logic should be put in the `fit` method. See [developing scikit-learn estimators](https://scikit-learn.org/stable/developers/develop.html#apis-of-scikit-learn-objects) for details.

Moreover, a test to ensure that `order` can be `list`, `tuple` and `ndarray` would be required.